Convert IDs to `String`

SpikeHD opened the pull request:
Fixes #7 and #10 

Converts all IDs to `String` in order to fix Navidrome compatibility.

SpikeHD committed:
fix: ids are strings

SpikeHD committed:
fix: album artist ID

xeals commented:
I suspect these were integers to start with because the most common server was Airsonic, which AFAICT still uses integers. It's been a while since I looked at the music server ecosystem and my gut feeling is that Navidrome is more popular these days, but I'm hesitant to swap one compatibility issue for another. If you're interested, I think a better solution would be to add a newtype wrapper around the ID and a custom deserializer that handles both cases.

SpikeHD committed:
Merge branch 'master' into fix/string_ids

SpikeHD commented:
@xeals That makes sense! As I was shaking my fist at Airsonic for not following spec I've implemented a `Id` enum that can be `String` or `Numeric`, and updated the codebase to use this. Consumers of the library should now be able to do this:

```rs
let my_song = Song::get(&client, "some-awesome-id");
let my_other_song = Song::get(&client, 1122345);
```

(same with albums, artists, etc)

SpikeHD committed:
feat: Id intermediate type

SpikeHD committed:
fix: add file oops
